### PR TITLE
Add kernels of SwooshForward and SwooshForwardAndDeriv

### PR DESCRIPTION
**Pull request by yaozengwei**
This PR adds cuda kernels of four functions: 
- SwooshLForward
- SwooshLForwardAndDeriv
- SwooshRForward
- SwooshRForwardAndDeriv

### CODE REVIEW

**Comment by danpovey** [comment]
Thanks.  BTW, if there is demand for it in the future, it might be nice to create a small separate repository for this so that people can get it without the rest of k2.  But no hurry on that.

File: k2/python/csrc/torch/v2/autograd/swoosh.cu [hunk 1]
```diff
@@ -183,6 +183,93 @@ using SwooshRFunction = SwooshFunction<SwooshRConstants>;
...
+// simple version of SwooshL that does not redefine the backprop, used in
+// ActivationDropoutAndLinearFunction.
+template <typename SwooshConstants>
+torch::Tensor SwooshForward(torch::Tensor x) {
+  static constexpr float kShift = SwooshConstants::kShift;
+  static constexpr float kCoeff = SwooshConstants::kCoeff;
+  static constexpr float kOffset = SwooshConstants::kOffset;
+
+  x = x.to(torch::kFloat32).contiguous();
+  float *x_data = x.data_ptr<float>();
```
Comment (csukuangfj) [inline k2/python/csrc/torch/v2/autograd/swoosh.cu:195]:
```suggestion
  const float *x_data = x.data_ptr<float>();
```

File: k2/python/csrc/torch/v2/autograd/swoosh.cu [hunk 2]
```diff
@@ -183,6 +183,93 @@ using SwooshRFunction = SwooshFunction<SwooshRConstants>;
...
+
+  torch::Tensor y = torch::empty_like(x).contiguous();
+  float *y_data = y.data_ptr<float>();
+
+  float shift = kShift;
+  float coeff = kCoeff;
+  float offset = kOffset;
+  auto context = GetContext(x);
+  DeviceGuard guard(context);
```
Comment (csukuangfj) [inline k2/python/csrc/torch/v2/autograd/swoosh.cu:204]: Please move these two lines to the beginning of this function.

File: k2/python/csrc/torch/v2/autograd/swoosh.cu [hunk 3]
```diff
@@ -183,6 +183,93 @@ using SwooshRFunction = SwooshFunction<SwooshRConstants>;
...
+// avoid infinity / infinity = nan.
+template <typename SwooshConstants>
+std::tuple<torch::Tensor, torch::Tensor> SwooshForwardAndDeriv(
+    torch::Tensor x) {
+  static constexpr float kShift = SwooshConstants::kShift;
+  static constexpr float kCoeff = SwooshConstants::kCoeff;
+  static constexpr float kOffset = SwooshConstants::kOffset;
+
+  x = x.to(torch::kFloat32).contiguous();
+  float *x_data = x.data_ptr<float>();
```
Comment (csukuangfj) [inline k2/python/csrc/torch/v2/autograd/swoosh.cu:238]:
```suggestion
  const float *x_data = x.data_ptr<float>();
```

Comment (yaozengwei) [reply k2/python/csrc/torch/v2/autograd/swoosh.cu:238]: Thanks.

**Comment by csukuangfj** [comment]
Thanks!

**Pull request merged**